src/DonorProfiles/Admin/UpgradeNotice.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Give\DonorProfiles\Admin;
+
+class UpgradeNotice {
+
+	public function register() {
+		if ( $this->shouldRenderOutput() ) {
+			$this->renderOutput();
+		}
+	}
+
+	protected function shouldRenderOutput() {
+
+		// Give Admin Only.
+		if ( give_is_admin_page() ) {
+
+			$donorProfilePageIsSet = empty( give_get_option( 'donor_profile_page' ) ) || get_post_status( give_get_option( 'donor_profile_page' ) ) === false ? false : true;
+			$historyPageIsSet      = empty( give_get_option( 'history_page' ) ) ? false : true;
+
+			if ( $donorProfilePageIsSet === false && $historyPageIsSet === true ) {
+				return true;
+			} else {
+				return false;
+			}
+			return true;
+		} else {
+			return false;
+		}
```

Comment:
@henryholtgeerts I suggest a return early method, in general. It helps avoid nesting. Also, the final `return true` inside the conditional block is unreachable.

Finally, `empty($foo) ? false : true` and `if ($truthy) { return true; } else { return false; }` are redundant. The `empty()` function returns a boolean, and conditional expressions return booleans — so just used the returned booelan.
```suggestion
		if ( ! give_is_admin_page() ) {
			return false;
		}

		$donorProfilePageIsSet = ! empty( give_get_option( 'donor_profile_page' ) ) && get_post_status( give_get_option( 'donor_profile_page' ) );
		$historyPageIsSet      = ! empty( give_get_option( 'history_page' ) );

		return ! $donorProfilePageIsSet && $historyPageIsSet;
```

Reply:
Ah, good thinking! Much clearer. I've gone ahead and reflected this change in other areas where I attempt something similar: https://github.com/impress-org/givewp/pull/5660/commits/3f2ad24b7e1c6428160f9b16518d93379e7f38ec